Signal/src/Models/TSMessageAdapaters/AttachmentPointerAdapter.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import UIKit
+
+/**
+ * Represents a download-in-progress
+ */
+class AttachmentPointerAdapter: JSQMediaItem, OWSMessageEditing {
+
+    let TAG = "[AttachmentPointerAdapter]"
+    let isIncoming: Bool
+    let attachmentPointer: TSAttachmentPointer
+    var cachedView: UIView?
+    var attachmentPointerView: AttachmentPointerView?
+
+    required init(attachmentPointer: TSAttachmentPointer, isIncoming: Bool) {
+        self.isIncoming = isIncoming
+        self.attachmentPointer = attachmentPointer
+        super.init(maskAsOutgoing: !isIncoming)
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        assertionFailure("init(coder:) has not been implemented")
+        self.isIncoming = true
+        self.attachmentPointer = TSAttachmentPointer()
+        super.init(coder: aDecoder)
+    }
+
+    func canPerformAction(_ action: Selector) -> Bool {
+        // No actions can be performed on a downloading attachment.
+        return false
+    }
+
+    func performAction(_ action: Selector) {
+        // Should not get here, as you can't perform any actions on a downloading attachment.
+        Logger.error("\(TAG) unexpectedly trying to perform action: \(action) on downloading attachment.")
+        assertionFailure()
+    }
+
+    override func mediaViewDisplaySize() -> CGSize {
+        return CGSize(width: 200, height: 90)
+    }
+
+    override func mediaView() -> UIView? {
+        guard self.cachedView == nil else {
+            return self.cachedView
+        }
+
+        let frame = CGRect(origin: CGPoint.zero, size: self.mediaViewDisplaySize())
+        let view = UIView(frame: frame)
+        self.cachedView = view
+
+        JSQMessagesMediaViewBubbleImageMasker.applyBubbleImageMask(toMediaView: view, isOutgoing:!isIncoming)
+
+        view.isUserInteractionEnabled = false
+
+        let attachmentPointerView = AttachmentPointerView(attachmentPointer: attachmentPointer, isIncoming: self.isIncoming)
+        self.attachmentPointerView = attachmentPointerView
+        view.addSubview(attachmentPointerView)
+
+        attachmentPointerView.autoPinWidthToSuperview(withMargin: 20)
+        attachmentPointerView.autoVCenterInSuperview()
```

Comment:
So `AttachmentPointerView` is guaranteed to always be smaller than `view`? Should we have `view` clip to bounds?